### PR TITLE
Fixed build documentation

### PR DESCRIPTION
**Pull request by ihorserba**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -213,9 +213,19 @@ Windows: Install autotools using [MSYS2](http://www.msys2.org/)
 pacman -S make autoconf automake libtool pkgconfig
 ```
 
-[Install Node.js](https://nodejs.org/en/download/)
+[Install Node.js 16 LTS](https://nodejs.org/dist/v16.13.2/node-v16.13.2.pkg)
```
Comment (vocoded) [inline README.md]: We could recommend `nvm` here as well

File: README.md [hunk 2]
```diff
@@ -213,9 +213,19 @@ Windows: Install autotools using [MSYS2](http://www.msys2.org/)
...
+Note: Node 17 does not support the required openssl functionality.
 
 [Install EMScripten](https://kripken.github.io/emscripten-site/docs/getting_started/downloads.html) v1.38.48
+```bash
+git clone https://github.com/emscripten-core/emsdk.git
+cd emsdk
+git checkout tags/1.38.48
+./emsdk install sdk-1.38.48-64bit
+./emsdk activate sdk-1.38.48-64bit
+source emsdk_env.sh
```
Comment (vocoded) [inline README.md:233]: is there an equivalent procedure for windows?

Comment (ihorserba) [reply README.md:233]: The windows build instructions are outdated, as [MSYS2 does not support Node.js anymore](https://github.com/msys2/MINGW-packages/issues/5848#issuecomment-539405968).

File: README.md (hunk 1 above)
Comment (ihorserba) [reply README.md]: Added recomendation

File: README.md (hunk 2 above)
Comment (ihorserba) [reply README.md:233]: Noted that currently windows build is not supported.
We have to find another way to build it.

**Comment by ihorserba** [comment]
@vocoded please take a look one more time - this PR has been forgotten for a while.

File: README.md [hunk 3]
```diff
@@ -208,14 +208,30 @@ Mac: Install autotools using [MacPorts](https://www.macports.org/install.php)
...
 ```bash
 pacman -S make autoconf automake libtool pkgconfig
 ```
 
-[Install Node.js](https://nodejs.org/en/download/)
+[Install Node.js 16 LTS](https://nodejs.org/dist/v16.13.2/node-v16.13.2.pkg)
+
+Use [nvm](https://github.com/nvm-sh/nvm#install--update-script) in order to quickly install and switch between the different Node.js versions.
+
+Note: Node 17 does not support the required openssl functionality.
```
Comment (vocoded) [inline README.md:224]: Can we bump to node 18 which is LTS?

**Review by vocoded — APPROVED** [review]
changes are ok, one question since we moved the SDK to node 18 in the other PR

File: README.md (hunk 3 above)
Comment (ihorserba) [reply README.md:224]: Unfortunately, we can't.
There's an [ERR_OSSL_EVP_UNSUPPORTED error](https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported).

When setting the `openssl-legacy-provider` option - I've got an error from emscripten:
```
/Users/iserba/dev/emsdk/node/12.9.1_64bit/bin/node: --openssl-legacy-provider is not allowed in NODE_OPTIONS
```
So downgrading to the node v16 is the only way I found

Comment (ihorserba) [reply README.md:224]: Unfortunately, we can't.
There's an [ERR_OSSL_EVP_UNSUPPORTED error](https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported).

When setting the `openssl-legacy-provider` option - I've got an error from emscripten:
```
/Users/iserba/dev/emsdk/node/12.9.1_64bit/bin/node: --openssl-legacy-provider is not allowed in NODE_OPTIONS
```
So downgrading to the node v16 is the only way I found

**Comment by ihorserba** [comment]
Finally got rid of the warnings:
```
$ npm install

added 228 packages, and audited 229 packages in 8s

38 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```
```
$ npm run make
...
  54 passing (540ms)
```

@vocoded - could you take a look one more time please?

File: README.md [hunk 4]
```diff
@@ -208,14 +208,30 @@ Mac: Install autotools using [MacPorts](https://www.macports.org/install.php)
...
 ```
 
-Windows: Install autotools using [MSYS2](http://www.msys2.org/)
+Windows: The [MSYS2](http://www.msys2.org/) does not support Node.js anymore. We have to find another way to build it using the Windows platform.
+
+The old (obsolete) instructions:
+
+Install autotools using [MSYS2](http://www.msys2.org/)
 ```bash
 pacman -S make autoconf automake libtool pkgconfig
```
Comment (adambailey-) [inline README.md:217]: These instructions don't work on newer macOS
```
nadehz@Adams-MacBook-Pro-5 opus-recorder % port install automake autoconf libtool pkgconfig

Error: Current platform "darwin 21" does not match expected platform "darwin 19"
Error: If you upgraded your OS, please follow the migration instructions: https://trac.macports.org/wiki/Migration
OS platform mismatch
    while executing
"mportinit ui_options global_options global_variations"
Error: /opt/local/bin/port: Failed to initialize MacPorts, OS platform mismatch
```

File: package.json [hunk 5]
```diff
@@ -36,16 +36,20 @@
     "test": "make test",
     "webpack": "webpack"
   },
+  "engines": {
+    "node": ">=10.13.0 <=16",
```
Comment (adambailey-) [inline package.json]: why are we supporting so many node versions? wouldn't it be better to just lock it into newer node LTS?

File: README.md (hunk 4 above)
Comment (ihorserba) [reply README.md:217]: I believe that's a problem of the previous installation of the [macports](https://www.macports.org/install.php) on the macOS before it is being upgraded. So you have to either reinstall macports or perform the [migration procedure](https://trac.macports.org/wiki/Migration)

File: package.json (hunk 5 above)
Comment (ihorserba) [reply package.json]: I'm not sure about the best practices for an SDK and related things.
In my understanding - if somebody includes the SDK code to his own project - the more node versions are supported, it's more likely the project's node version can be used.
However when thinking again - the SDK users have to use the compiled SDK, right?
In that case it's definitely better to lock to the exact version.

Comment (ihorserba) [reply package.json]: Created a [separate PR](https://github.com/zelloptt/opus-recorder/pull/8) for this.
Please take a look

Comment (ihorserba) [reply package.json]: Merged

File: README.md (hunk 3 above)
Comment (vocoded) [reply README.md:224]: we're good to go with node v16, thanks

**Review by vocoded — APPROVED** [review]
Assuming @adambailey- can validate on his setup, changes look fine to me

**Comment by ihorserba** [comment]
> Assuming @adambailey- can validate on his setup, changes look fine to me

@adambailey- could you please confirm?

**Comment by ihorserba** [comment]
Per discussion with @adambailey - reduced the debug level from `-g3` to `-g2` of the unminified WebAssembly binaries in order to avoid generation of the `wast` artifacts.

Rebuilt the binaries and updated the package-lock file

FYI:
https://emscripten.org/docs/tools_reference/emcc.html

`-g2`: When linking, preserve function names in compiled code.
`-g3`: When compiling to object files, keep debug info, including JS whitespace, function names, and LLVM debug info (DWARF) if any (this is the same as [-g](https://emscripten.org/docs/tools_reference/emcc.html#emcc-g))

**Review by adambailey- — APPROVED** [review]
Tested and built, looks good to me, thanks!

**Pull request merged**